src/test/scala/chiselTests/AsTypeOfTester.scala

```diff
@@ -120,34 +122,559 @@ class AsChiselEnumTester extends BasicTester {
+        )
+        for (test <- tests) {
+          val verilog = ChiselStage.emitSystemVerilog(test())
+          verilog should include("wire [7:0] out = in;")
+        }
+      }
+    }
+    describe("as the source type") {
+      they("should expand narrower inputs") {
+        val verilog = ChiselStage.emitSystemVerilog(new SimpleAsTypeOf(UInt(8.W), UInt(10.W)))
+        verilog should include("wire [9:0] out = {2'h0, in};")
+      }
+      they("should truncate wider inputs") {
+        val verilog = ChiselStage.emitSystemVerilog(new SimpleAsTypeOf(UInt(8.W), UInt(6.W)))
+        verilog should include("wire [5:0] out = in[5:0];")
+      }
+      they("should match for matched width inputs") {
+        val verilog = ChiselStage.emitSystemVerilog(new SimpleAsTypeOf(UInt(8.W), UInt(8.W)))
+        verilog should include("wire [7:0] out = in;")
+      }
+      they("should work for inferred width outputs") {
+        val verilog = ChiselStage.emitSystemVerilog(new SimpleAsTypeOf(UInt(8.W), UInt()))
+        verilog should include("wire [7:0] out = in;")
+      }
+    }
   }
 
-  it should "work for casting to and from ChiselEnums" in {
-    assertTesterPasses(new AsChiselEnumTester)
+  describe("Analogs") {
+    describe("as the target type") {
+      they("should error") {
+        val e = the[ChiselException] thrownBy ChiselStage.emitSystemVerilog(new RawModule {
+          val in = IO(Input(UInt(8.W)))
+          val out = IO(Analog(8.W))
+          out := in.asTypeOf(out)
+        })
+        e.getMessage should include("Analog does not support connectFromBits")
+      }
+    }
+    describe("as the source type") {
+      they("should error") {
+        val e = the[ChiselException] thrownBy ChiselStage.emitSystemVerilog(new RawModule {
+          val in = IO(Analog(8.W))
+          val out = IO(Output(UInt(8.W)))
+          out := in.asTypeOf(out)
+        })
+        e.getMessage should include("Analog does not support asUInt")
+      }
+    }
   }
 
-  it should "work for casting to and from Clock" in {
-    assertTesterPasses(new AsTypeOfClockTester)
+  describe("Bools") {
+    describe("as the target type") {
+      they("should expand narrower inputs") {
+        val tests = List(
+          () => new SimpleAsTypeOf(UInt(0.W), Bool()),
+          () => new WireSourceAsTypeOf(UInt(), Bool())(_ := 0.U(0.W))
+        )
+        for (test <- tests) {
+          val verilog = ChiselStage.emitSystemVerilog(test())
+          verilog should include("wire out = 1'h0;")
+        }
+      }
+      they("should truncate wider inputs") {
+        val tests = List(
+          () => new SimpleAsTypeOf(UInt(8.W), Bool()),
+          () => new WireSourceAsTypeOf(UInt(), Bool())(_ := 0.U(8.W))
+        )
+        for (test <- tests) {
+          val verilog = ChiselStage.emitSystemVerilog(test())
+          // 8-bit wire for inferred test puts several spaces between wire and out
+          (verilog should include).regex("""wire +out = in\[0\];""")
```

Comment:
This type of test is going to be somewhat annoyingly brittle. You could probably skip the wire (probably in all tests, too).